tests/unittest/test_breakpad_resource.py

```diff
@@ -367,13 +360,93 @@ def text_extract_payload_kvpairs_and_json(self, request_generator, metricsmock):
             method="POST", path="/submit", headers=headers, body=data
         )
 
-        bsp = BreakpadSubmitterResource(config=self.empty_config, crashmover=crashmover)
+        bsp = BreakpadSubmitterResource(
+            config=EMPTY_CONFIG, crashmover=FakeCrashMover()
+        )
         with metricsmock as metrics:
             result = bsp.extract_payload(req)
             assert result == ({}, {})
             assert metrics.has_record(stat="malformed", tags=["reason:has_json_and_kv"])
 
+
+@pytest.mark.parametrize(
+    "raw_crash, expected",
+    [
+        ({}, {"metadata": {"collector_notes": []}}),
+        (
+            {"TelemetryClientId": "ou812"},
+            {
+                "metadata": {
+                    "collector_notes": ["Removed TelemetryClientId from raw crash."]
+                }
+            },
+        ),
+        (
+            {"TelemetryServerURL": "ou812"},
+            {
+                "metadata": {
+                    "collector_notes": ["Removed TelemetryServerURL from raw crash."]
+                }
+            },
+        ),
+    ],
+)
+def test_cleanup_crash_report(raw_crash, expected):
```

Comment:
This was part of the monolithic `TestBreakpadSubmitterResource`. I broke that up into several groups. This test and the next test are better at the top level on their own.